### PR TITLE
Sample Chaincode for Asset lifecycle management for Bond and Fungible Asset.

### PR DESCRIPTION
**Pull request by sanvenDev**
1. Added chaincode for BondAsset and FungibleAsset with tests.
2. Fixed the bug in testnet with wrong ca-org port in network2.

### CODE REVIEW

File: samples/simpleasset/bondasset.go [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+func (s *SmartContract) AssetExists(ctx contractapi.TransactionContextInterface, id string) (bool, error) {
+	assetJSON, err := ctx.GetStub().GetState(id)
+	if err != nil {
+		return false, fmt.Errorf("failed to read from world state: %v", err)
+	}
+
+	return assetJSON != nil, nil
+}
+// AssetExists returns true when asset with given ID exists in world state
+func (s *SmartContract) isAssetReleased(ctx contractapi.TransactionContextInterface, id string) (bool, error) {
```
Comment (VRamakrishna) [inline samples/simpleasset/bondasset.go:116]: Can you capitalize the function name? `IsAssetReleased`?

File: samples/simpleasset/fungibleasset.go [hunk 2]
```diff
@@ -0,0 +1,266 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/hyperledger/fabric-contract-api-go/contractapi"
+)
+
+type FungibleAssetType struct {
```
Comment (VRamakrishna) [inline samples/simpleasset/fungibleasset.go:10]: I suggest changing this to something more specific, like `TokenAsset`.

File: samples/simpleasset/fungibleasset.go [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+	Issuer                string            `json:"issuer"`
+	FaceValue             int               `json:"facevalue"`
+}
+type Wallet struct {
```
Comment (VRamakrishna) [inline samples/simpleasset/fungibleasset.go:14]: Perhaps call this `TokenWallet`?

File: samples/simpleasset/fungibleasset.go [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+// Checks if owner has some given amount of fungible asset
+func (s *SmartContract) FungibleAssetsExist(ctx contractapi.TransactionContextInterface, fungibleAssetType string, numUnits int, owner string) (bool, error) {
+	balance, err := s.GetBalance(ctx, fungibleAssetType, owner)
+	if err != nil {
+		return false, err
+	}
+	return balance >= numUnits, nil
+}
+
+func addFungibleAssetsHelper(ctx contractapi.TransactionContextInterface, id string, fungibleAssetType string, numUnits int) error {
```
Comment (VRamakrishna) [inline samples/simpleasset/fungibleasset.go:194]: Capitalize `add`.

File: samples/simpleasset/mocks/statequeryiterator.go [hunk 5]
```diff
@@ -0,0 +1,232 @@
+// Code generated by counterfeiter. DO NOT EDIT.
```
Comment (VRamakrishna) [inline samples/simpleasset/mocks/statequeryiterator.go:1]: Do you need to check these `mocks` files in? I thought you were writing a `Makefile` to import them on the fly from Fabric Interop CC folder?

File: samples/simpleasset/main.go [hunk 6]
```diff
@@ -0,0 +1,43 @@
+package main
+
+import (
+  "fmt"
+  "os"
+
+  "github.com/hyperledger/fabric-chaincode-go/shim"
+	"github.com/hyperledger/fabric-contract-api-go/contractapi"
```
Comment (VRamakrishna) [inline samples/simpleasset/main.go:8]: Can you fix indentations in the imports so they are vertically aligned?

File: samples/simpleasset/fungibleasset.go (hunk 4 above)
Comment (sanvenDev) [reply samples/simpleasset/fungibleasset.go:194]: That is a helper function, not a member function of Smart Contract. Should I still capitalize that?

**Review by VRamakrishna — APPROVED** [review]

**Pull request merged**